docs/adr/0011-operators.md

```diff
@@ -0,0 +1,153 @@
+
+According to the upstream documentation on [`OperatorGroup` target namespace selection][] we enforce and support only these options for target namespace specification in _Operator groups_ specification:
+
+1. A global, cluster-wide operator with visibility to all namespaces, that means the `.spec` section of the manifest is blank.
+2. A single namespace operator. Resource manifest specifies `.spec.targetNamespaces: [ SINGLE_MEMBER_NAMESPACE ]`. This namespace has to be deployed via `cluster-scope` application as well.
+
+The `OperatorGroup` resource manifest is expected to be deployed to the same namespace where the corresponding `Subscription` is located.
+
+### Direct operator deployment
+
+Some operators are not deployed via the OLM. That means, their manifests require manual deployment of `CustomResourceDefinition`s, `ClusterRole`s etc. We require these kind of resources to be deployed via `cluster-scope` application.
+
+In contrast to that, all the other resources required for a non-OLM operator deployment, like `Deployment`, `Route` etc. are treated as any other namespace scoped application and all the manifests required for deployment are expected to be located as a separate application in the `operate-first/apps` repository as a root folder.
+
+### Meta-operators
+
+This is a special kind of operator. A meta-operator in this context is understood to be any operator allowing deployment of other operators at cluster scope. Our intention is to limit the possibilities of such operator to be able to interfere with our cluster management in `cluster-scope` application. We expect from meta-operators to provide options to either:
+
+- Limit their permissions via operator configuration, so they are not able to deploy resources requiring cluster-admin permissions anymore.
+- Limit the create/edit access permission to the custom resources they create, in order to restrict normal project users or project admins to escalate their privileges.
+
+If it is not possible to deploy a meta-operator in a way that respects one of the 2 conditions, it is expected from the operator requester to raise this issue upstream. Until this is fixed we expect the operator to be installed manually via [Direct operator deployment](#direct-operator-deployment), not via OLM framework, therefore allowing us to manually control the custom resource access permissions for regular users or the operator service account `ClusterRole`.
+
+### Positive Consequences
+
+- Clear guidance on each and individual resource regarding operators, which may appear confusing without such rules.
+- Allows operator to be treated like normal applications in case they are deployed manually. This will not clutter the `cluster-scope` applicaton anymore.
+- Enforces demands on meta-operators that makes it hard to escalate user privileges via meta-operators.
```

Comment:
this is strange: Enforces demands ... that makes. It sounds strange and mixes plural and singular.

Reply:
Perhaps:

> Enforces restrictions that prevent user privilege escalation via meta-operators.